runtime/drivers/clickhouse/olap.go

```diff
@@ -250,7 +250,57 @@ func (c *connection) InsertTableAsSelect(ctx context.Context, name, sql string,
 			LongRunning: true,
 		})
 	}
-
+	if strategy == drivers.IncrementalStrategyReplace {
+		// create temp table with the same schema
+		tempName := tempName(name)
+		// drop the temp table
+		defer func() {
+			err := c.DropTable(ctx, tempName, false)
+			if err != nil {
+				c.logger.Error("clickhouse: failed to drop temp table", zap.String("name", tempName), zap.Error(err))
+			}
+		}()
+		err := c.Exec(ctx, &drivers.Statement{
```

Comment:
Use `CREATE OR REPLACE`.
Is it guaranteed to work for distributed table in all case ? it is missing `on cluster clause` ?

Reply:
Yes. that works on distributed table. This just creates an empty table with same schema. 

See https://clickhouse.com/docs/en/sql-reference/statements/create/table#with-a-schema-similar-to-other-table

>  it is missing on cluster clause

We can't pass cluster info here. It will be created on same cluster as existing table.

Reply:
Distributed is an abstract table => there is a base table => it is possible to manipulate on the base table. Thanks @esevastyanov